extensions/qute/runtime/src/main/java/io/quarkus/qute/runtime/devmode/QuteErrorPageSetup.java

```diff
@@ -0,0 +1,223 @@
+
+    private static final String TEMPLATE_EXCEPTION = "io.quarkus.qute.TemplateException";
+    private static final String ORIGIN = "io.quarkus.qute.TemplateNode$Origin";
+
+    private static final String PROBLEM_TEMPLATE = ""
+            + "<h3>#{problemIndex} {title}</h3>\n"
+            + "<div style=\"margin-bottom:0.5em;\">{description}</div>\n"
+            + "<div style=\"font-family:monospace;font-size:1em;background-color:#2E3436;color:white;padding:1em;margin-bottom:2em;\">\n"
+            + "{#if realLines.get(0) > 1}<span style=\"color:silver;\">...</span><br>{/if}\n"
+            + "{#for line in sourceLines}\n"
+            // highlight the error line - start
+            + "{#if lineNumber is realLines.get(index)}<div style=\"background-color:#555753;\">{/if}\n"
+            // line number
+            + "<span style=\"color:silver;\">{realLines.get(index).pad}</span>\n"
+            // line content
+            + " {line}\n"
+            // highlight the error line - end 
+            + "{#if lineNumber is realLines.get(index)}</div>{#else}<br>{/if}\n"
+            // point to error
+            + "{#if lineNumber is realLines.get(index)}{space.pad}<span style=\"color:red;\">{#for i in lineCharacterStart}={/for}^</span><br>{/if}\n"
+            + "{/for}\n"
+            + "{#if endLinesSkipped}<span style=\"color:silver;\">...</span>{/if}\n"
+            + "</div>";
+
+    private HotReplacementContext hotReplacementContext;
+
+    @Override
+    public void setupHotDeployment(HotReplacementContext context) {
+        this.hotReplacementContext = context;
+        ErrorPageGenerators.register(TEMPLATE_EXCEPTION, this::generatePage);
+    }
+
+    String generatePage(Throwable exception) {
+        Escaper escaper = Escaper.builder().add('"', "&quot;").add('\'', "&#39;")
+                .add('&', "&amp;").add('<', "&lt;").add('>', "&gt;").build();
+        Template problemTemplate = Engine.builder().addDefaults().addValueResolver(new ReflectionValueResolver())
+                .addValueResolver(new ValueResolver() {
+
+                    public boolean appliesTo(EvalContext context) {
+                        return context.getName().equals("pad");
+                    }
+
+                    @Override
+                    public CompletionStage<Object> resolve(EvalContext context) {
+                        return CompletableFuture.completedFuture(htmlPadRight(context.getBase().toString(), 5));
+                    }
+                })
+                .build()
+                .parse(PROBLEM_TEMPLATE);
+        TemplateHtmlBuilder builder;
+        List<Throwable> problems;
+        Throwable[] suppressed = exception.getSuppressed();
+        if (suppressed.length == 0) {
+            problems = Collections.singletonList(exception);
+        } else {
+            problems = Arrays.asList(suppressed);
+        }
+
+        String problemsFound = "Found " + problems.size() + " Qute problems";
```

Comment:
Can you please sort the errors by increasing location? You example screenshot shows the last error shown first and it's confusing :)

Reply:
Done.